Lib/typing.py

```diff
@@ -2020,14 +2024,15 @@ def _proto_hook(cls, other):
                 break
 
             # ...or in annotations, if it is a sub-protocol.
-            if (
-                issubclass(other, Generic)
-                and getattr(other, "_is_protocol", False)
-                and attr in _lazy_annotationlib.get_annotations(
-                    base, format=_lazy_annotationlib.Format.FORWARDREF
-                )
-            ):
-                break
+            if issubclass(other, Generic) and getattr(other, "_is_protocol", False):
+                try:
```

Comment:
```suggestion
                # We avoid the slower path through annotationlib here because in most
                # cases it should be unnecessary.
                try:
```